policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/ApplicationTestData.java

```diff
@@ -45,52 +41,40 @@
             .id(RESOURCE_ID)
             .type(RESOURCE_TYPE)
             .serialisedFormat(RESOURCE_FORMAT)
             .connectionDetail(new SimpleConnectionDetail().serviceName(DATA_SERVICE_NAME))
             .parent(new SystemResource().id(RESOURCE_PARENT));
+
     public static final String PURPOSE = "test-purpose";
     public static final Context CONTEXT = new Context().purpose(PURPOSE);
     public static final String RULE_MESSAGE = "test-rule";
-    public static final Rules<Serializable> RULES = new Rules<Serializable>().addRule(RULE_MESSAGE, new PassThroughRule<Serializable>());
+
+    public static final Rules<LeafResource> RESOURCE_RULES = new Rules<LeafResource>().addRule(RULE_MESSAGE, new PassThroughRule<>());
+    public static final Rules<Serializable> RULES = new Rules<>().addRule(RULE_MESSAGE, new PassThroughRule<>());
+
     public static final PolicyRequest REQUEST = PolicyRequest.Builder.create()
             .withUserId(USER_ID.getId())
             .withResourceId(RESOURCE_ID)
             .withContext(CONTEXT)
             .withUser(USER)
             .withResource(LEAF_RESOURCE);
     public static final PolicyResponse RESPONSE = PolicyResponse.Builder.create(REQUEST)
-            .withResource(LEAF_RESOURCE)
             .withRules(RULES);
-    public static final ProducerRecord<String, PolicyRequest> START = new ProducerRecord<>("resource", 0, null, null);
-    public static final ProducerRecord<String, PolicyRequest> RECORD = new ProducerRecord<>("resource", 0, null, REQUEST);
-    public static final ProducerRecord<String, PolicyRequest> END = new ProducerRecord<>("resource", 0, null, null);
-    // Create a stream of resources, uniquely identifiable by their type, which is their position in the stream (first resource has type "0", second has type "1", etc.)
-    public static final Supplier<Stream<ProducerRecord<String, PolicyRequest>>> RECORD_FACTORY = () -> Stream.iterate(0, i -> i + 1)
-            .map(i -> PolicyRequest.Builder.create()
-                    .withUserId(USER_ID.getId())
-                    .withResourceId(RESOURCE_ID)
-                    .withContext(CONTEXT)
-                    .withUser(USER)
-                    .withResource(LEAF_RESOURCE))
-            .map(request -> new ProducerRecord<>(
-                    "resource",
-                    0,
-                    (String) null,
-                    request,
-                    RECORD.headers()));
 
-    static {
-        START.headers().add(StreamMarker.HEADER, StreamMarker.START.toString().getBytes());
-        START.headers().add(Token.HEADER, REQUEST_TOKEN.getBytes());
+    public static final AuditErrorMessage AUDIT_ERROR_MESSAGE = AuditErrorMessage.Builder.create()
+            .withUserId(USER_ID.getId())
+            .withResourceId(RESOURCE_ID)
+            .withContext(CONTEXT)
+            .withAttributes((new HashMap<>()))
+            .withError(new Throwable("Something went wrong!"));
```

Comment:
Do you want this to be a policy exception in line with your other change?

Reply:
Done